session.go

```diff
@@ -213,20 +248,25 @@ func (s *session) sendLoop(pingInterval time.Duration) {
 	lastPing := time.Now()
 
 	onFrame := func(frame []byte) {
+		// Reserve space for header in sessionFrame
+		startOfData = 2
+
 		// Coalesce pending writes. This helps with performance and blocking
 		// resistence by combining packets.
 		coalescedBytes = 0
 		coalesced := 1
 		if s.clientInitMsg != nil {
-			// Lazily send client init message with first data
-			copy(coalesceBuffer, s.clientInitMsg)
-			coalescedBytes += clientInitSize
-			coalesced++
+			// Lazily send client init message with first data, but don't encrypt
+			copy(sessionFrame, s.clientInitMsg)
+			// Push start of data right
+			startOfData += clientInitSize
 			s.clientInitMsg = nil
 		}
 		bufferFrame(frame)
+
 	coalesceLoop:
-		for coalescedBytes < coalesceThreshold {
+		// Coalesce as much as possible without exceeding maxSessionFrameSize
+		for startOfData+coalescedBytes+maxHMACSize+maxFrameSize < maxSessionFrameSize {
```

Comment:
So `maxHMACSize+maxFrameSize` is the maximum bytes of one encrypted frame right? So it depends on the `dataEncrypter` to use, though both AEAD implementations have an overhead of 16. It's better to be derived from the `cryptoSpec` when `startSession`.